atomic_reactor/plugins/post_koji_upload.py

```diff
@@ -0,0 +1,642 @@
+            all_images = pod.get_container_image_ids()
+        except OsbsException as ex:
+            self.log.error("unable to find image id: %r", ex)
+            return buildroot_tag
+
+        try:
+            return all_images[buildroot_tag]
+        except KeyError:
+            self.log.error("Unable to determine buildroot image ID for %s",
+                           buildroot_tag)
+            return buildroot_tag
+
+    def get_buildroot(self, build_id):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: dict, partial metadata
+        """
+
+        docker_info = self.tasker.get_info()
+        host_arch, docker_version = get_docker_architecture(self.tasker)
+
+        buildroot = {
+            'id': 1,
+            'host': {
+                'os': docker_info['OperatingSystem'],
+                'arch': host_arch,
+            },
+            'content_generator': {
+                'name': PROG,
+                'version': atomic_reactor_version,
+            },
+            'container': {
+                'type': 'docker',
+                'arch': os.uname()[4],
+            },
+            'tools': [
+                {
+                    'name': tool['name'],
+                    'version': tool['version'],
+                }
+                for tool in get_version_of_tools()] + [
+                {
+                    'name': 'docker',
+                    'version': docker_version,
+                },
+            ],
+            'components': self.get_rpms(),
+            'extra': {
+                'osbs': {
+                    'build_id': build_id,
+                    'builder_image_id': self.get_builder_image_id(),
+                }
+            },
+        }
+
+        return buildroot
+
+    def get_logs(self):
+        """
+        Build the logs entry for the metadata 'output' section
+
+        :return: list, Output instances
+        """
+
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = self.osbs.get_build_logs(self.build_id)
+        except OsbsException as ex:
+            self.log.error("unable to get build logs: %r", ex)
+        else:
+            # Deleted once closed
+            logfile = NamedTemporaryFile(prefix=self.build_id,
+                                         suffix=".log",
+                                         mode='wb')
+            logfile.write(logs.encode('UTF-8'))
+            logfile.flush()
+            metadata = self.get_output_metadata(logfile.name,
+                                                "openshift-final.log")
+            output.append(Output(file=logfile, metadata=metadata))
+
+        docker_logs = NamedTemporaryFile(prefix="docker-%s" % self.build_id,
+                                         suffix=".log",
+                                         mode='wb')
+        docker_logs.write("\n".join(self.workflow.build_result.logs).encode('UTF-8'))
```

Comment:
I hand merged commit 9e09d607d1c5b648fbbc92224ec300b930d0945e, though I am not sure what the added test does, as it doesn't verify anything.